tools/clang/unittests/HLSLExec/ExecutionTest.cpp

```diff
@@ -11096,6 +11228,609 @@ TEST_F(ExecutionTest, PackUnpackTest) {
+  }
+
+  VERIFY_IS_TRUE(clampArgsC.size() == clampArgsT.size(), L"ClampArgs vectors must be the same size.");
+
+  CComPtr<IStream> pStream;
+  ReadHlslDataIntoNewStream(L"ShaderOpArith.xml", &pStream);
+
+  const size_t numElementsPerVector = vecInput1[0].size();
+
+  // Create the compiler args outside of the lambda so the string pointer stays
+  // valid for the lifespan of the test method. Really should just use something
+  // like a std::shared_ptr<std::string>
+  char compilerOptions[256];
+  std::string type("float");
+  VERIFY_IS_TRUE(sprintf_s(compilerOptions, sizeof(compilerOptions),
+                           "-DTYPE=%s -D NUM=%zu", type.c_str(), numElementsPerVector) != -1);
+
+  std::shared_ptr<ShaderOpTestResult> test = RunShaderOpTest(
+    pDevice, m_support, pStream, "LongVectorUnaryOp", 
+    // this callback is called when the test is creating the resource to run
+    // the test. We use it to override the values set on the st:ShaderOp from
+    // ShaderOpArith.xml. We update with values defined in ShadoerOpArithTable.xml.
+    [&](LPCSTR Name, std::vector<BYTE> &Data, st::ShaderOp *pShaderOp) {
+      LogCommentFmt(L"RunShaderOpTest CB. Shader Name: %S", Name);
+      VERIFY_IS_TRUE(0 == _stricmp(Name, "SLongVectorUnaryOp<T>"));
+
+      const size_t size = sizeof(SLongVectorUnaryOp<T, N>) * inputVectorCount;
+      Data.resize(size);
+
+      // Treat the data as an array of SLongVectorUnaryOp<T, N> structs.
+      // One struct for each set of <value> tags in the  ShaderOpArithTable.xml
+      // A set of value tags is the corresponding <value> tags across <Parameters>
+      SLongVectorUnaryOp<T, N> *pPrimitives = (SLongVectorUnaryOp<T, N> *)Data.data();
+
+      for (size_t i = 0; i < inputVectorCount ; ++i) {
+        SLongVectorUnaryOp<T, N> *p = &pPrimitives[i];
+
+        std::copy(vecInput1[i].begin(), vecInput1[i].end(), p->vecInput1.begin());
+
+        if(!clampArgsC.empty())
+        {
+          VERIFY_IS_TRUE(clampArgsC[i].size() == 1, L"ClampArgs vectors must have exactly one element.");
+          VERIFY_IS_TRUE(clampArgsT[i].size() == 1, L"ClampArgs vectors must have exactly one element.");
+          p->clampArgC = clampArgsC[i][0];
+          p->clampArgT = clampArgsT[i][0];
+        }
+      }
+
+      // Override the shader name and text (source hlsl) with the values from ShaderOpArithTable.xml
+      pShaderOp->Shaders.at(0).Target = Target.m_psz;
+      pShaderOp->Shaders.at(0).Text = Text.m_psz;
+
+      // And override the compiler arguments. We pass in the element
+      // type for the vectors and their size.
+      pShaderOp->Shaders.at(0).Arguments = compilerOptions;
+    }
+  );
+
+  // Map the data from GPU to CPU memory so we can verify our expectations.
+  MappedData data;
+  test->Test->GetReadBackData("SLongVectorUnaryOp<T>", &data);
+
+  // Cast the buffer back into an array of the structs we expect.
+  SLongVectorUnaryOp<T, N> *pPrimitive = (SLongVectorUnaryOp<T, N>*)data.data();
```

Comment:
There's a lot of new code being added here in a mix of style conventions. Ideally new code should conform to the LLVM conventions (see: [the variable naming conventions](https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly)), unless there is a consistent style in the existing code.

For this you have a lot of new code which is itself inconsistent. We should avoid that.

Also I have a particular dislike of Hungarian notation, which is why I left this comment on this line specifically.

Reply:
The coding styles in these tests are all over the place. Auto-complete in my editor was having a bad time because of it. I intended to come back on a formatting pass. 

They're also already using Hungarian notation in most places.

Reply:
Yeah, when we're adding new code in a file like this that's inconsistent we follow llvm naming conventions.  Although the llvm doc doesn't explictly say "no hungarian", I think it is pretty clear that hungarian is incompatible with the naming conventions described.